spec/middleware-1.0.0-draft.1.md

```diff
@@ -0,0 +1,309 @@
+material or to cite them other than as "work in progress"
+
+This working draft will expire on 1 March 2016.
+
+## Copyright
+
+Copyright (c) 2014 the persons identified as the document authors. The work
+is licensed as per the
+[Creative Commons Attribution 3.0 Unported License][CCLicense] in effect on the
+date of publication of this document.
+
+## Table of Content
+
+1. [Introduction](#1-introduction)  
+  1.1. [Requirements Notation](#11-requirements-notation)
+2. [Definitions](#2-definitions)
+3. [Middleware](#3-middleware)  
+  3.1. [Examples](#31-examples)  
+  3.2. [Middleware Operation](#32-middleware-operation)  
+  3.3. [Signature](#33-signature)  
+  3.4. [Code Sample](#34-code-sample)
+4. [Middleware Registration](#4-middleware-registration)  
+  4.1. [Signature](#41-signature)  
+    4.1.1. [Registration Sample](#411-registration-sample)  
+  4.2. [Discovery and Registration of Middleware](#42-discovery-and-registration-of-middleware)  
+    4.2.1. [Extension Method Sample](#421-extension-method-sample)  
+5. [Versioning](#5-versioning)
+6. [Acknowledgements](#6-acknowledgements)
+
+## 1. Introduction
+
+The [OWIN] specification was created to decouple web servers and web
+applications on .NET platforms by creating a standardised abstract signature
+that can be implemented by Applications and Servers in a vendor-neutral fashion.
+
+[Layering][FieldingLayering] components conforming to a uniform interface
+is a natural architectural style for HTTP-based application, and many
+application frameworks provide such abstractions in the form of reusable
+software components.
+
+This specification provides a standardised signature for such components in the
+OWIN ecosystem, hereby referred to as **Middleware**.
+
+OWIN Middleware is defined in terms of a delegate structure building upon the
+OWIN: Open Web Server Interface for .NET [Owin 1.0], with no
+dependency on external libraries.
+
+### 1.1. Requirements Notation
+
+ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119][rfc2119].
+
+A component is deemed to be compliant with this specification if all absolute
+requirements and prohibitions are conformed to.
+
+Components and terms defined in this and other specifications are always
+capitalized, so as to clarify between a general term and the term as used in a
+specification.
+
+## 2. Definitions
+
+This document refers to the following terms:
+
+ - **Application**: a process or operation inspecting, modifying or otherwise
+   accessing HTTP messages, as an AppFunc signature.
+
+ - **Middleware**: Pass through components, capable of mediating request and
+   response messages, in a pipeline model which, when
+   composed, forms an Application.
+
+ - **Builder**: The extension point lambda used by developers to add Middleware
+   components to a pipeline.
```

Comment:
I used Middleware and Middleware components as synonyms here, chosing Middleware most of the time, unless it just sounds a bit wrong. "may add Middleware" just sounds a bit wrong, even if it's accurate. I'm not fussed either way. What do people think?